### PR TITLE
[CU-jd835e] Creator view - add scheduling settings

### PR DESCRIPTION
**Pull request by Buglik**
Capture1
Capture2
Capture3
Capture4
Capture5
Capture6
Capture7
Capture8

### CODE REVIEW

**Comment by Zummek** [comment]
Task linked: [CU-jd835e Creator view - add scheduling settings](https://app.clickup.com/t/jd835e)

File: web-app/src/components/tournament/creator/DaysOfPlaySelector.vue [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+
+@Component
+export default class DaysOfPlaySelector extends Vue {
+  @Prop({ type: Array, required: true })
+  days!: [];
+
+  private pickedDays = this.days;
```
Comment (Zummek) [inline web-app/src/components/tournament/creator/DaysOfPlaySelector.vue:36]: If you want, u can make that through SyncProp or PropSync. You can find an example of that in our project.
As a result, I think we can get less code

File: web-app/src/components/tournament/creator/FrequencySelector.vue [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+@Component
+export default class Frequency extends Vue {
+  @Prop({ type: Number, required: true })
+  frequency!: number;
+
+  private newFreq = this.frequency;
+
+  updateDays() {
+    this.$emit('update-freq', this.newFreq);
+  }
```
Comment (Zummek) [inline web-app/src/components/tournament/creator/FrequencySelector.vue]: Same suggestion as above(SyncProp)

File: web-app/src/components/tournament/creator/DaysOfPlaySelector.vue (hunk 1 above)
Comment (Zummek) [reply web-app/src/components/tournament/creator/DaysOfPlaySelector.vue:36]: https://vuejs.org/v2/guide/components-custom-events.html#sync-Modifier

File: web-app/src/components/tournament/creator/StartDateSelector.vue [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+@Component
+export default class StartDateSelector extends Vue {
+  @Prop({ type: String, required: true })
+  date!: string;
+
+  private newDate = this.date;
+
+  updateDate() {
+    this.$emit('update-date', this.newDate);
+  }
```
Comment (Zummek) [inline web-app/src/components/tournament/creator/StartDateSelector.vue]: same

File: web-app/src/components/tournament/creator/TournamentNameChanger.vue [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+@Component
+export default class TournamentNameChanger extends Vue {
+  @Prop({ type: String, required: true })
+  tournamentName!: string;
+
+  private newTournName = this.tournamentName;
+
+  updateTournamentName() {
+    this.$emit('name-changed', this.newTournName);
+  }
```
Comment (Zummek) [inline web-app/src/components/tournament/creator/TournamentNameChanger.vue]: same

File: web-app/src/components/tournament/creator/TournamentTypeSelectorItem.vue [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
   <div class="row col-8 col-sm-4 justify-center q-ma-md">
-    <q-btn class="row col-12" no-caps padding="none">
+    <q-btn class="row col-12 selectorButton" no-caps padding="none">
       <q-card
         flat
-        class="col-12 tournamentTypeCard"
+        class="row col-12 tournamentTypeCard items-center"
         :class="tournamentType === activeType ? 'activeTypeCard' : ''"
         @click="updateTournamentType"
       >
```
Comment (Zummek) [inline web-app/src/components/tournament/creator/TournamentTypeSelectorItem.vue:9]: When did you set height/width through `selectorButton` to `q-btn` is required to set this same to child component `q-card`?

File: web-app/src/pages/tournament/TournamentCreator.vue [hunk 6]
```diff
@@ -225,6 +253,10 @@ export default class TournamentCreator extends Vue {
   }
 
   private async submitAddTournament() {
+    const dateToSubmit = new Date(Date.parse(this.startDateString));
+    // console.log(dateToSubmit);
+    // console.log(this.days);
+    // console.log(this.frequency)
```
Comment (Zummek) [inline web-app/src/pages/tournament/TournamentCreator.vue]: .

File: web-app/src/components/tournament/creator/TournamentTypeSelectorItem.vue (hunk 5 above)
Comment (Buglik) [reply web-app/src/components/tournament/creator/TournamentTypeSelectorItem.vue:9]: When not set, q-card doesn't take full btn space, pic below 
obraz

File: web-app/src/pages/tournament/TournamentCreator.vue (hunk 6 above)
Comment (Buglik) [reply web-app/src/pages/tournament/TournamentCreator.vue]: leaving comments for now, these are not connected to api yet

**Review by eastemployeer — APPROVED** [review]

**Pull request merged**